src/App.tsx

```diff
@@ -1,54 +1,105 @@
-import React from 'react';
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+
+// eslint-disable-next-line max-len
+const debounce = (f: { (value: React.SetStateAction<string>): void; (...args: [string]): void; }, delay: number | undefined) => {
+  let timerId: NodeJS.Timeout;
+
+  return (...args: [string]) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(f, delay, ...args);
+  };
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [closeList, setCloseList] = useState(false);
+
+  const visiblePeople = useMemo(() => {
+    // eslint-disable-next-line no-console
+    console.log('Filtering', appliedQuery);
+
+    return peopleFromServer.filter(
+      (person) => {
+        return person.name.includes(appliedQuery);
+      },
+    );
+  }, [peopleFromServer, appliedQuery]);
```

Comment:
The dependency array should only contain 'appliedQuery', not 'peopleFromServer'.